Tensile/KernelWriterAssembly.py

```diff
@@ -1113,28 +1113,27 @@ def initKernel(self, kernel, tPA, tPB ):
 
     # registers per element
     self.bpr = 4 # all registers are 32bit
-    self.bpeAB = int(self.bpr*\
-        kernel["ProblemType"]["DataType"].numRegisters())
-    self.bpeCexternal = int(self.bpr*\
-        kernel["ProblemType"]["DataType"].numRegisters())
+    self.bpeAB = int(self.bpr * kernel["ProblemType"]["DataType"].numRegisters())
+    self.bpeCexternal = int(self.bpr * kernel["ProblemType"]["DataType"].numRegisters())
+    self.bpeDexternal = int(self.bpr * kernel["ProblemType"]["DataType"].numRegisters())
```

Comment:
bpeCexternal / bpeDexternal should use kernel["ProblemType"]["DestDataType"] ? Treat as though A/B may be different type than C/D ?

Reply:
Also what is the difference between bpeCexternal and bpeDexternal? Do we support different types of C / D ?

Reply:
Hi @cgmillette,
We have a special case FP16 high precision accumulation.
Which compute datatype is FP16, but accumulation type is FP32.
I believe this is a legacy issue.
So we have to use bpeCexternal / bpeDexternal instead of use kernel["ProblemType"]["DestDataType"].
I hope we can correct special case FP16 and remove high precision accumulation, but we don't

In our HPA BF16 GSU algorithm, we use workspace which type is FP32 for temporarily D and C buffer which type is FP16.
So we need bpeCexternal / bpeDexternal for C & D.